components/suggest/src/db.rs

```diff
@@ -1058,6 +1115,51 @@ impl<'a> SuggestDao<'a> {
         Ok(())
     }
 
+    /// Inserts exposure suggestion records data into the database.
+    pub fn insert_exposure_suggestions(
+        &mut self,
+        record_id: &SuggestRecordId,
+        suggestion_type: &str,
+        suggestions: &[DownloadedExposureSuggestion],
+    ) -> Result<()> {
+        // `suggestion.keywords()` can yield duplicates for exposure
+        // suggestions, so ignore failures on insert in the uniqueness
+        // constraint on `(suggestion_id, keyword)`.
+        let mut keyword_insert = KeywordInsertStatement::new_with_or_ignore(self.conn)?;
+        let mut suggestion_insert = SuggestionInsertStatement::new(self.conn)?;
+        for suggestion in suggestions {
+            self.scope.err_if_interrupted()?;
+            let suggestion_id = suggestion_insert.execute(
+                record_id,
+                "",
+                "",
+                DEFAULT_SUGGESTION_SCORE,
+                SuggestionProvider::Exposure,
+            )?;
+            self.conn.execute_cached(
+                "INSERT INTO exposure_custom_details(
+                     suggestion_id,
+                     type
+                 )
+                 VALUES(
+                     :suggestion_id,
+                     :type
+                 )",
+                named_params! {
+                    ":suggestion_id": suggestion_id,
+                    ":type": suggestion_type,
+                },
+            )?;
```

Comment:
Please add a  `.with_context` call before checking the error type.  That way if we get SQL errors, we'll see which statement caused them.  Secondarily, I think it would faster to define an `ExposureInsertStatement` type like the others.  I think `execute_cached` does a map lookup each iteration.  This is much less important though.